pymdownx/inlinehilite.py

```diff
@@ -126,7 +126,7 @@ def get_settings(self):
             self.css_class = css_class if css_class else config['css_class']
 
             self.extend_pygments_lang = config.get('extend_pygments_lang', None)
-            self.guess_lang = config['guess_lang']
+            self.guess_lang = (config['guess_lang'] == True) or (config['guess_lang'] == 'inline')
```

Comment:
The Highlight class itself should be the only thing that processes the input. It is the only thing that should be aware.  We instantiate the Highlight class in InlineHilite only when it hasn't already been included as it is dependent on it, but the logic should be contained in Highlight.

Reply:
Fixed.

On a side note, I think InlineHilite instantiates its own `Highlight` regardlessly.

https://github.com/facelessuser/pymdown-extensions/blob/ea60333049783280dd90b304a670da12ff853a67/pymdownx/inlinehilite.py#L139-L150

Reply:
It does. It was modeled after SuperFences which allows a few inline overrides via fence code options, such as line numbers on a per block basis, but generally, handling of that logic is performed in the Highlight class.

There was some uncertainty as to whether we would need a per inline override. I've worked to avoid it.